### PR TITLE
Build product listing page

### LINKED ISSUE #4: Todo: Build product listing page
In this issue will create the product listing page and the styling of the pages.

### PR DESCRIPTION
**Pull request by vaibhav-mahalle**
this pull request contains:

- changes in the UI of the product-listing page.
- added the sort by price functionality.
- added filterContext and filterReducer.
- added the favicon for the app.

Screenshot (113)

Screenshot (114)

### CODE REVIEW

**Review by nakulsharma15 — APPROVED** [review]
Looks good to me. No changes suggested. I Liked the way you structured the files in their respective folders and the names which you've given to the variables and functions

File: src/components/index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import Header from "./Header";
 import Footer from "./Footer";
 import ProductCard from "./ProductCard";
+import Filter from "./Filter";
```
Comment (rahulcs754) [inline src/components/index.js:4]: you can set here export no need to write line number 6

**Review by rahulcs754 — APPROVED** [review]
The code looks good and clean
just small suggestion

File: src/components/index.js (hunk 1 above)
Comment (vaibhav-mahalle) [reply src/components/index.js:4]: I will make the required changes.

File: src/index.js [hunk 2]
```diff
@@ -3,15 +3,18 @@ import ReactDOM from "react-dom";
...
 // Call make Server
 makeServer();
 
 ReactDOM.render(
   <React.StrictMode>
     <Router>
-      <App />
+      <FilterProvider>
+        <App />
+      </FilterProvider>
```
Comment (sakleprajwal) [inline src/index.js:17]: I am not sure why are you using <App /> two times here.
I think all the context providers should be used in App.js page. The index.js page should look clean.

**Review by sakleprajwal — COMMENTED** [review]
I have provided one suggestion from my end. Everything else looks good. Please go ahead

**Pull request merged**